crowdsourcing/permissions/project.py

```diff
@@ -13,3 +14,19 @@ def has_object_permission(self, request, view, obj):
         if request.method in permissions.SAFE_METHODS:
             return True
         return obj.worker.profile.user == request.user
+
+
+class IsProjectAvailable(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        query = '''
+                SELECT wr.requester_id as id FROM get_worker_ratings(%(worker_profile)s) wr
```

Comment:
it seems a bit not efficient to fetch all ratings and then pick one, maybe another function which does a more specific lookup could be more efficient? i.e only get the ratings by this requester for this worker, same goes for project min_ratings